src/mydumper_start_dump.c

```diff
@@ -822,62 +836,44 @@ void send_lock_all_tables(MYSQL *conn){
       }
 
       g_string_printf(
-            query,
-            "SELECT TABLE_SCHEMA, TABLE_NAME FROM information_schema.TABLES "
-            "WHERE TABLE_SCHEMA in (%s) AND TABLE_TYPE ='BASE TABLE' AND NOT "
-            "(TABLE_SCHEMA = 'mysql' AND (TABLE_NAME = 'slow_log' OR "
-            "TABLE_NAME = 'general_log'))",
-            db_quoted_list->str);
+        query, 
+        "SELECT TABLE_SCHEMA, TABLE_NAME FROM information_schema.TABLES "
+        "WHERE TABLE_SCHEMA in (%s) AND TABLE_TYPE ='BASE TABLE'",
+        db_quoted_list->str);
     } else {
       g_string_printf(
         query,
         "SELECT TABLE_SCHEMA, TABLE_NAME FROM information_schema.TABLES "
         "WHERE TABLE_TYPE ='BASE TABLE' AND TABLE_SCHEMA NOT IN "
-        "('information_schema', 'performance_schema', 'data_dictionary') "
-        "AND NOT (TABLE_SCHEMA = 'mysql' AND (TABLE_NAME = 'slow_log' OR "
-        "TABLE_NAME = 'general_log'))");
+        "('information_schema', 'performance_schema', 'data_dictionary')");
     }
-  }
-
-  if (tables_lock == NULL && query->len > 0  ) {
     if (mysql_query(conn, query->str)) {
       g_critical("Couldn't get table list for lock all tables: %s",
                  mysql_error(conn));
       errors++;
     } else {
-      MYSQL_RES *res = mysql_store_result(conn);
-      MYSQL_ROW row;
-
+      res = mysql_store_result(conn);
       while ((row = mysql_fetch_row(res))) {
-        lock = 1;
-        if (tables) {
-          int table_found = 0;
-          for (i = 0; tables[i] != NULL; i++)
-            if (g_ascii_strcasecmp(tables[i], row[1]) == 0)
-              table_found = 1;
-          if (!table_found)
-              lock = 0;
-        }
-        if (lock && tables_skiplist_file && check_skiplist(row[0], row[1]))
+        // no need to check if the tb exists in the tables.
+        if (tables_skiplist_file && check_skiplist(dt[0], row[0]))
           continue;
-        if (lock && !eval_regex(row[0], row[1]))
```

Comment:
@shihuafan please, review my last commit about this change

Reply:
ok